packages/node/src/indexer/indexer.manager.ts

```diff
@@ -175,11 +180,21 @@ export class IndexerManager {
 
   async start(): Promise<void> {
     await this.projectService.init();
+    logger.info('indexer manager started');
   }
 
-  private filterDataSources(nextProcessingHeight: number): SubqlProjectDs[] {
+  private async filterDataSources(
+    nextProcessingHeight: number,
+  ): Promise<SubqlProjectDs[]> {
     let filteredDs: SubqlProjectDs[];
-    filteredDs = this.project.dataSources.filter(
+    if (!this.modifiedDataSources) {
+      this.modifiedDataSources =
+        await generateTimestampReferenceForBlockFilters(
```

Comment:
I think this could be called in `ProjectService.init` instead and we can update the datasources on `this.project.dataSources` directly. Then we have no need to `modifiedDatasources`.